core/src/main/java/hudson/views/JobColumn.java

```diff
@@ -36,6 +39,24 @@
     public JobColumn() {
     }
 
+    public String columnCaption(View view) {
+        String metaName = null;
+        for (Item item : view.getItems()) {
+            String m = item instanceof AbstractItem
+                    ? ((AbstractItem) item).getMetaName()
+                    : Messages.JobColumn_DisplayName();
```

Comment:
return immediately? If there are only non-<code>AbstractItem</code> items, the logic will iterate through all jobs without any chance to change the result